libsolidity/parsing/Parser.cpp

```diff
@@ -291,19 +291,23 @@ ASTPointer<ContractDefinition> Parser::parseContractDefinition()
 			Token currentTokenValue = m_scanner->currentToken();
 			if (currentTokenValue == Token::RBrace)
 				break;
-			else if (currentTokenValue == Token::Function || currentTokenValue == Token::Constructor)
-				// This can be a function or a state variable of function type (especially
-				// complicated to distinguish fallback function from function type state variable)
-				subNodes.push_back(parseFunctionDefinitionOrFunctionTypeStateVariable());
+			else if (
```

Comment:
You seem do be doing a bit of refactoring here in the Parser, could you explain what the changes are? I see you remove a lot of the `parseFunctionDefinition` code and I guess you handle that somewhere else..

Reply:
*`parseFunctionHeader` I mean

Reply:
Having to distinguish ``function() external x;`` (state variable) from ``function() external {}`` (fallback) caused quite a few complications in the parser, which now aren't necessary anymore - you can distinguish variables from functions simply by checking for ``function`` + identifier vs. ``function`` + ``(``. That should account for all changes in the parser.